### PR TITLE
Allow prop "disabled" passed from filter to menuItem

### PR DESCRIPTION
**Pull request by ZachSelindh**
This PR allows filters to accept the prop "disabled," which is passed to the MUI MenuItem component, which is already set up to accept "disabled" as a part of ButtonBase props.

### CODE REVIEW

**Review by djhi — CHANGES_REQUESTED** [review]
Hi and thanks for opening a PR. Would you mind updating the documentation and updating the tests?

**Comment by ZachSelindh** [comment]
> Hi and thanks for opening a PR. Would you mind updating the documentation and updating the tests?

Just pushed a simple filter menu test and sentence added to the docs specifying that filter menu items/inputs can be disabled.

File: docs/List.md [hunk 1]
```diff
@@ -942,6 +942,8 @@ const postFilters = [
 
 Elements of the `filters` array are regular inputs. That means you can build sophisticated filters based on references, array values, etc. `<List>` hides all inputs in the filter form by default, except those that have the `alwaysOn` prop.
 
+Filters will render as disabled inputs or menu items (depending on filter context) if passed the prop "disabled."
```
Comment (WiXSL) [inline docs/List.md:945]:
```suggestion
**Tip**: Filters will render as disabled inputs or menu items (depending on filter context) if passed the prop "disabled".
```

**Review by WiXSL — CHANGES_REQUESTED** [review]

**Review by WiXSL — COMMENTED** [review]
Please, resolve file conflicts

**Review by djhi — APPROVED** [review]

**Comment by WiXSL** [comment]
Test are failing

**Comment by ZachSelindh** [comment]
> Test is failing

I'm unsure how to provide a substantive test for this functionality without adding a testing dependency, @testing-library/jest-dom

I'm open to suggestions if you're not interested in adding the dependency.

**Comment by WiXSL** [comment]
> > Test is failing
> 
> I'm unsure how to provide a substantive test for this functionality without adding a testing dependency, @testing-library/jest-dom
> 
> I'm open to suggestions if you're not interested in adding the dependency.

We have to ask @fzaninotto about this

File: packages/ra-ui-materialui/src/list/filter/FilterButton.tsx [hunk 2]
```diff
@@ -84,6 +84,7 @@ export const FilterButton = (props: FilterButtonProps): JSX.Element => {
                         filter={filterElement}
                         resource={resource}
                         onShow={handleShow}
+                        disabled={filterElement.props.disabled || false}
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/list/filter/FilterButton.tsx:87]: I don't understand why you set the disabled prop here, it has no effect?

File: packages/ra-ui-materialui/src/list/filter/FilterButtonMenuItem.tsx [hunk 3]
```diff
@@ -39,12 +40,14 @@ FilterButtonMenuItem.propTypes = {
     filter: PropTypes.element.isRequired,
     onShow: PropTypes.func.isRequired,
     resource: PropTypes.string,
+    disabled: PropTypes.bool,
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/list/filter/FilterButtonMenuItem.tsx:43]: you've added a prop but you don't use it...

**Review by fzaninotto — CHANGES_REQUESTED** [review]

**Comment by fzaninotto** [comment]
> I'm unsure how to provide a substantive test for this functionality without adding a testing dependency, @testing-library/jest-dom

Can you elaborate? What' can't you do with the current testing setup?

File: packages/ra-ui-materialui/src/list/filter/FilterButton.tsx (hunk 2 above)
Comment (ZachSelindh) [reply packages/ra-ui-materialui/src/list/filter/FilterButton.tsx:87]: Agreed, on second look this is unnecessary. Will delete this change

**Comment by ZachSelindh** [comment]
> > I'm unsure how to provide a substantive test for this functionality without adding a testing dependency, @testing-library/jest-dom
> 
> Can you elaborate? What' can't you do with the current testing setup?

Thanks for taking a look at this! It may very well be my inexperience with writing these tests, but it seems that the method "toHaveAttribute" from jest-dom testing library is the conventional way of testing if a MUI MenuItem renders as disabled, see below:

https://stackoverflow.com/questions/64553059/how-to-check-if-a-material-ui-menuitem-is-disabled-using-react-testing-library

**Comment by fzaninotto** [comment]
> Thanks for taking a look at this! It may very well be my inexperience with writing these tests,

Right, if you just copy and paste StackOverflow code, you're in for surprises!

The current test setup, involving react-testing-library, is more than enough for this test. Please learn how to use it and add a test using it.

**Comment by ZachSelindh** [comment]
Used SelectInput and SelectArrayInput tests as a template, getting the attribute "aria-disabled".

File: docs/List.md [hunk 4]
```diff
@@ -700,6 +702,3 @@ const PostList = () => (
 );
 ```
 {% endraw %}
-
-**Tip**: The `List` component `classes` can also be customized for all instances of the component with its global css name `RaList` as [describe here](https://marmelab.com/blog/2019/12/18/react-admin-3-1.html#theme-overrides)
```
Comment (fzaninotto) [inline docs/List.md:704]: why did you remove this section?

**Review by WiXSL — APPROVED** [review]

File: docs/List.md (hunk 4 above)
Comment (ZachSelindh) [reply docs/List.md:704]: @fzaninotto Apologies, that was a mistake from a merge from a previous branch. Fixed now!

**Comment by fzaninotto** [comment]
Thanks a lot!
